arkouda/message.py

```diff
@@ -54,6 +58,129 @@ def dict(self):
+    @staticmethod
+    @typechecked
+    def _build_strings_param(key: str, val) -> ParameterObject:
+        """
+        Create a ParameterObject from a Strings value
+
+        Parameters
+        ----------
+        key : str
+            key from the dictionary object
+        val
+            Strings object ot load from the symbol table
+
+        Returns
+        -------
+        ParameterObject
+        """
+        # empty string if name of String obj is none
+        name = val.name if val.name else ""
+        return ParameterObject(key, ObjectType.STRINGS, "str", name)
+
+    @staticmethod
+    @typechecked
+    def _build_list_param(key: str, val: list) -> ParameterObject:
+        """
+        Create a ParameterObject from a list
+
+        Parameters
+        ----------
+        key : str
+            key from the dictionary object
+        val : list
+            list object to format as string
+
+        Returns
+        -------
+        ParameterObject
+        """
+        dtypes = set([p.dtype for p in val])
+        if len(dtypes) > 1:
+            t_str = ", ".join(dtypes)
+            raise TypeError(f"List values must be of the same type. Found {t_str}")
+        return ParameterObject(key, ObjectType.LIST, dtypes.pop(), json.dumps(val))
+
+    @staticmethod
+    @typechecked
+    def _build_gen_param(key: str, val) -> ParameterObject:
+        """
+        Create a ParameterObject from a single value
+
+        Parameters
+        ----------
+        key : str
+            key from the dictionary object
+        val
+            singular value to use. This could be str, int, float, etc
+
+        Returns
+        -------
+        ParameterObject
+        """
+        v = val if isinstance(val, str) else str(val)
+        return ParameterObject(key, ObjectType.VALUE, type(val).__name__, v)
+
+    @staticmethod
+    def generate_dispatch() -> Dict:
+        """
+        Builds and returns the dispatch table used to build parameter object.
+
+        Returns
+        -------
+        Dictionary - mapping the parameter type to the build function
+        """
+        from arkouda.pdarrayclass import pdarray
+        from arkouda.strings import Strings
+
+        return {
+            pdarray.__name__: ParameterObject._build_pdarray_param,
+            Strings.__name__: ParameterObject._build_strings_param,
+            list.__name__: ParameterObject._build_list_param,
+        }
+
+    @staticmethod
+    def factory(key: str, val) -> ParameterObject:
+        """
+        Factory method used to build ParameterObject given a key value pair
+
+        Parameters
+        ----------
+        key : str
+            key from the dictionary object
+        val
+            the value corresponding to the provided key from the dictionary
+
+        Returns
+        --------
+        ParameterObject - The parameter object formatted to be parsed by the chapel server
+        """
+        dispatch = ParameterObject.generate_dispatch()
```

Comment:
is it possible to pull `dispatch` out, so we can cache the table instead of rebuilding? I first I thought we might be able to do just define it since it doesn't appear to change, but then i realized their might be circular dependency issues

Reply:
We probably can, I will look into this.

Reply:
In order to do this (due to the circular references) we would have to update the current return from `generate_dispatch()` to 
```
return {
           "pdarray": ParameterObject._build_pdarray_param,
            "Strings": ParameterObject._build_strings_param,
            list.__name__: ParameterObject._build_list_param,
        }
```
This is not really an issue to do, but will rely on devs providing the correct value for arkouda class names because it would need to be the equivalent to `class.__name__`.

Reply:
We would also need to change the structure because making it a class property of `ParameterObject` is not possible due to the fact that the dispatch is mapping to static methods within the class which are not yet defined. 

The other option would be to create the functions at the file level and have `arkouda.client.PARAM_DISPATCH`. This would again require the changes above unless we defined inside a function to prevent issues with the circular import. It would end up looking something like this
```
def generate_dispatch() -> Dict:
    """
    Builds and returns the dispatch table used to build parameter object.

    Returns
    -------
    Dictionary - mapping the parameter type to the build function
    """
    from arkouda.pdarrayclass import pdarray
    from arkouda.strings import Strings

    return {
        pdarray.__name__: _build_pdarray_param,
        Strings.__name__: _build_strings_param,
        list.__name__: _build_list_param,
    }
PARAM_DISPATCH = generate_dispatch()
```
This would also require the `_build_*` functions to be moved out of the `ParameterObject` class

Reply:
Attempted configuring as above and we still get stuck with the circular import issues.

Reply:
@pierce314159 definitely makes an excellent point. However, given the circular dependency stuff here, I recommend keeping this as-is for now and adding a tech debt issue to resolve the circular dependency